### ISSUE TITLE
Decouple plugin-manager from napari

### ISSUE THREAD

**Issue by jaimergp** [napari/napari-plugin-manager#59]
Other dev teams have expressed interest in using our plugin manager in their ecosystem. To that end, we offered to provide a plugin-manager UI that is uncoupled from napari as much as possible. The end goal would be a `napari-plugin-manager` that depends and extends a hypothetical, application agnostic `plugin-manager-dialog` (name tbd).

## To do:

> These will become their own items as needed 

- [ ] Investigate coupled areas and design API for `plugin-manager-dialog`
- [ ] Refactor `napari-plugin-manager` into two packages (still in the same repo) following the design proposed above. These two steps can happen iteratively (implementation informs design and viceversa)
- [ ] Decide maintenance and release model: two repositories, fully independent? One repository, coupled releases? Single repo but uncoupled releases?

**Comment by dalthviz** [comment]
Gave this an initial check and from the UI part (widgets definitions under `qt_plugin_dialog` for the most part) got a list of methods that use/have napari related logic:

* Custom rendering for plugins items:
	* `PluginListItem` class: (`setup_ui`, `_handle_npe2_plugin`, `_on_enabled_checkbox`)
* Plugins list:
	* `QPluginList` class: (`handle_action`, `tag_outdated`, `tag_unavailable`)
* Plugins dialog:
	* `QtPluginDialog` class: (`_fetch_available_plugins`, `_add_to_installed`, `_add_installed`, `_handle_yield`, `_update_theme`, `_setup_shortcuts`, `setup_ui`, `_on_installer_all_finished`, `_add_items`)

Also, there are some widgets that are being used which come from napari like `QtToolTipLabel`. Maybe their usage/creation should be encapsualted on a method to override, right?

Some other general elements that probably should be encapsulated into a method to override/should be marked with something like `NotImplementedError` is the translations handling (when widgets display text it is being first passed to the napari `trans._` function).

Could it make sense to start doing small PRs (like to encapsulate the translation logic or the `QtToolTipLabel` usage) or maybe I should give a more in depth check to the code first? Also, before start doing any big PR changing things, probably any pending major work related with the UI should be merged? Like #51? What do you think @jaimergp ?

**Comment by jaimergp** [comment]
Thank you for the analysis, Daniel!

I agree that ideally #51 would go in first, but maybe we can start small already, by proposing the base class + subclass split? If @goanpeca is not available this week, you can also push to #51 and finish it if you think that's easier.

**Comment by jaimergp** [comment]
> However, other implementations could extend the base class and make the query method to query things from a file or define a constant/variable with the info needed without the need of querying a web service.

Yep, that's what I had in mind too. napari happens to have the npe2api service (and we can document what we are doing there if someone wants to reuse the same approach for their stuff), but in principle users should define their own list population logic. In the future, if folks are really interested in the npe2api-like stuff, we could think of providing a 2nd level base class that streamlines the API querying but I think that's a second step, not the first one.

**Comment by dalthviz** [comment]
Continuing with the work here, I was checking how things would look dividing the code into two packages. However, while doing that, I noticed that the `qt_package_installer` module, although defines some level of abstraction, seems like has still some level of coupling with napari releated things (mostly around constrains definitions/file generation). Should I try to divide that module also into a `base_qt_package_installer` for the base clases and a `qt_package_installer` for napari related logic?

Also, thinking on the two packages division, there are a couple of PRs open which probably should be updated and merged before proceeding with the package division, right? 

What do you think @jaimergp ?
